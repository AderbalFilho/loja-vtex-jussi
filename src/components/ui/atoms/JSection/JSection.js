import React from 'react';
import classNames from 'classnames/bind';
import PropTypes from 'prop-types';

import './JSection.scss';

function JSection({ children, classes, color, ...rest }) {
  const className = classNames('j-section', `j-section--${color}`, classes);

  return (
    <section className={className} {...rest}>
      {children}
    </section>
  );
}

JSection.defaultProps = {
  color: 'white',
};

JSection.propTypes = {
  children: PropTypes.node,
  classes: PropTypes.string,
  color: PropTypes.oneOf(['white', 'black', 'pink', 'green']),
};

export default JSection;
