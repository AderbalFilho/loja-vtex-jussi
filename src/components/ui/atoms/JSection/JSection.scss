.j-section {
  width: 100%;
}

.j-section--white {
  background-color: var(--white);
}

.j-section--black {
  background-color: var(--black);
}

.j-section--pink {
  background-color: var(--pink-jussi);
}

.j-section--green {
  background-color: var(--green-jussi);
}
