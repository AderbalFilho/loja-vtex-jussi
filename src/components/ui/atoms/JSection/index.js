import JSection from './JSection';

export default JSection;
