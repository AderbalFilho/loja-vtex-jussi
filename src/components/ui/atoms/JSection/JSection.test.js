import React from 'react';
import TestRenderer from 'react-test-renderer';
import { render } from '@testing-library/react';

import JSection from './JSection';

describe('JSection', () => {
  it('matches snapshot', () => {
    const testRenderer = TestRenderer.create(<JSection>Jüssi</JSection>);
    const jsonComponent = testRenderer.toJSON();

    expect(jsonComponent).toMatchSnapshot();
  });

  describe('change color', () => {
    it('renders background-color white when no parameter is passed', () => {
      const { container } = render(<JSection>Jüssi</JSection>);
      const section = container.querySelector('.j-section');

      expect(section.classList).toContain('j-section--white');
    });

    it('renders background-color black', () => {
      const { container } = render(<JSection color="black">Jüssi</JSection>);
      const section = container.querySelector('.j-section');

      expect(section.classList).toContain('j-section--black');
    });

    it('renders background-color pink', () => {
      const { container } = render(<JSection color="pink">Jüssi</JSection>);
      const section = container.querySelector('.j-section');

      expect(section.classList).toContain('j-section--pink');
    });

    it('renders background-color green', () => {
      const { container } = render(<JSection color="green">Jüssi</JSection>);
      const section = container.querySelector('.j-section');

      expect(section.classList).toContain('j-section--green');
    });

    it('renders background-color white', () => {
      const { container } = render(<JSection color="white">Jüssi</JSection>);
      const section = container.querySelector('.j-section');

      expect(section.classList).toContain('j-section--white');
    });
  });

  it('renders children', () => {
    const { container } = render(<JSection>Jüssi</JSection>);
    const section = container.querySelector('.j-section');

    expect(section.textContent).toBe('Jüssi');
  });

  it('renders classes', () => {
    const { container } = render(<JSection classes="section1">Jüssi</JSection>);
    const section = container.querySelector('.j-section');

    expect(section.classList).toContain('section1');
  });
});
