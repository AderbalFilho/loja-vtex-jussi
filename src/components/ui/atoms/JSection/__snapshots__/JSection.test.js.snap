// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`JSection matches snapshot 1`] = `
<section
  className="j-section j-section--white"
>
  JÃ¼ssi
</section>
`;
